src/overlays/actors/ovl_player_actor/z_player.c

```diff
@@ -2357,12 +2374,26 @@ s32 Player_UpdateHostileLockOn(Player* this) {
     }
 }
 
-int func_80833BCC(Player* this) {
-    return Player_CheckHostileLockOn(this) || func_80833B2C(this);
+/**
+ * Returns true if currently Z-Targeting, false if not.
+ * Z-Targeting here is a blanket term that covers both the "actor lock-on" and "parallel" states.
+ *
+ * This variant of the function calls `Player_CheckHostileLockOn`, which does not update the hostile
+ * lock-on actor state.
+ */
+int Player_IsZTargeting(Player* this) {
+    return Player_CheckHostileLockOn(this) || Player_FriendlyLockOnOrParallel(this);
 }
 
-int func_80833C04(Player* this) {
-    return Player_UpdateHostileLockOn(this) || func_80833B2C(this);
+/**
+ * Returns true if currently Z-Targeting, false if not.
+ * Z-Targeting here is a blanket term that covers both the "actor lock-on" and "parallel" states.
+ *
+ * This variant of the function calls `Player_UpdateHostileLockOn`, which updates the hostile
+ * lock-on actor state before checking its state.
+ */
+int Player_IsZTargetingWithHostileUpdate(Player* this) {
```

Comment:
`Player_CheckZTargeting` and `Player_UpdateZTargeting` for these two maybe?

Reply:
its only updating the hostile state part. `func_80836BEC` is supposed to be UpdateZTargeting at a larger scale, which actually sets focusActor for lock on and puts you into parallel when appropriate

Reply:
I don't find the name `Player_IsZTargetingWithHostileUpdate` any clearer tbh (does it return true if link is z-targeting a hostile and also filling that hostile in on the latest gossip?). Is there any pattern in the usages of `Player_IsZTargetingWithHostileUpdate`? Maybe that can give some clues

Reply:
any clearer than what?
youll have to read the description of the function and `Player_UpdateHostileLockOn` which it calls to really understand, its not easy to convey this "update and also return the check" behavior in the name alone, at least I cant think of anything.

Instead of this thing being updated in a top level update function like it should, it is individual action function's responsibility to update the hostile state flag and also act against it. The non-update version gets used in non action functions.

Reply:
clearer than `Player_UpdateZTargeting` I mean (and renaming `func_80836BEC` to something else, it's less important for that to have a concise name since that's only called from one place). You would still have to check the description to see what it actually does, but at least it is less of a mouthful

Reply:
`Player_UpdateZTargeting` just isnt right though, the only thing being updated is one subset of many things that z targeting is. The higher level update function absolutely deserves that name

Reply:
I don't think that you're going to get much better than Player_IsZTargetingWithHostileUpdate for now. 
The way I look at it, Player_UpdateHostileLockOn is the same as Player_CheckHostileLockOn but with side effects, but I don't understand what situations you'd prefer one over the other since there still a large number of unnamed functions.

Reply:
agreed.
Basically you want to use the update version within action functions. Since this operation doesnt happen in an every-frame update call (like imo it should), you use that version in action functions. The non-update version is used in helper functions or other things that are not directly called from an action, and should not be updating the state flag.